app/change_set_persisters/plum_change_set_persister/apply_visibility_by_date.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+class PlumChangeSetPersister
+  class ApplyVisibilityByDate
+    attr_reader :change_set_persister, :change_set
+    def initialize(change_set_persister:, change_set:, post_save_resource: nil)
+      @change_set = change_set
+      @change_set_persister = change_set_persister
+    end
+
+    def run
+      return unless change_set.try(:set_visibility_by_date?)
+      return unless date_object
+      change_set.validate(visibility: calculated_visibility, rights_statement: calculated_rights_statement)
+      change_set.sync
+      change_set
+    end
+
+    private
+
+      def date
+        Array.wrap(change_set.resource.primary_imported_metadata.created).first
+      end
+
+      def date_object
+        return unless date
+        Time.zone.parse(date)
+      rescue
+        Rails.logger.debug("Unable to parse created date: #{change_set.resource.primary_imported_metadata.created}")
+        nil
+      end
+
+      def limit_date
+        Time.zone.parse("1924-01-01T00:00:00Z")
+      end
+
+      def public_access?
+        date_object < limit_date
+      end
+
+      def calculated_visibility
+        if public_access?
+          Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC
+        else
+          Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
+        end
+      end
+
+      def calculated_rights_statement
+        if public_access?
+          "http://rightsstatements.org/vocab/NKC/1.0/"
```

Comment:
We're using this as a default somewhere else, too. At what point to we consider loading these into constants? Perhaps a new ticket.

Reply:
Created #709